core/src/main/java/org/geysermc/geyser/extension/GeyserExtensionClassLoader.java

```diff
@@ -39,14 +40,19 @@
 
 public class GeyserExtensionClassLoader extends URLClassLoader {
     private final GeyserExtensionLoader loader;
+
+    private final ExtensionDescription description;
     private final Object2ObjectMap<String, Class<?>> classes = new Object2ObjectOpenHashMap<>();
 
-    public GeyserExtensionClassLoader(GeyserExtensionLoader loader, ClassLoader parent, Path path) throws MalformedURLException {
+    private boolean warned;
```

Comment:
```suggestion
    private boolean warnedForInternalClassLoad;
```
Makes it a bit more clear what exactly this means since "warned" is quite vague.